### PR TITLE
Add PumpkinPi board

### PR DESCRIPTION
**Pull request by tuftii**
First PR for PumpkinPi as mentionned in issue #679.

### CODE REVIEW

File: gpiozero/boards.py [hunk 1]
```diff
@@ -1684,3 +1684,63 @@ def off(self):
...
+    .. _ModMyPi PumpkinPi: https://www.modmypi.com/halloween-pumpkin-solder-kit
+
+    """
+    def __init__(self, pwm=False, initial_value=False, pin_factory=None):
+        super(PumpkinPi, self).__init__(
+            sides=LEDBoard(
+                left=LEDBoard(
+                    bottom=18, midbottom=17, middle=16, midtop=13, top=24,
+                    pwm=pwm, initial_value=initial_value,
+                    _order=('bottom','midbottom','middle','midtop','top'),
```
Comment (bennuttall) [inline gpiozero/boards.py]: Can we get a space after each comma in this tuple plz? (sorry)

Comment (bennuttall) [reply gpiozero/boards.py]: (same in other places)

File: gpiozero/__init__.py [hunk 2]
```diff
@@ -93,6 +93,7 @@
     PhaseEnableRobot,
     PololuDRV8835Robot,
     Energenie,
+    PumpkinPi,
```
Comment (bennuttall) [inline gpiozero/__init__.py]: Tests please

**Comment by bennuttall** [comment]
Forgot to say: just push more commits to the same branch and they'll be picked up here

**Comment by tuftii** [comment]
Yeah you're quite right @lurch, sorry I was updating both boards web pages yesterday and copied the wrong link.

**Comment by tuftii** [comment]
Ok, so I've added 3 tests for the PumpkinPi board. I came across something a bit weird.

I'm now setup with a development environment on Ubuntu using the mock pin environment variables. The first two tests (`test_pumpkin_pi` and `test_pumpkin_pi_initial_value`) run without issue.

The third `test_pumpkin_pi_initial_value_pwm` (which I've commented out for now) runs and passes, but causes `test_input_light_sensor`, `test_input_distance_sensor`, `test_mock_pin_frequency_unsupported` and `test_mock_pin_state` to fail. 

I'm a bit puzzled as to why. I've copied the test from the SnowPi pwm test and changed the pins to the PumpkinPi pins, but couldn't work out what would cause these other tests to fail?

File: gpiozero/__init__.py (hunk 2 above)
Comment (tuftii) [reply gpiozero/__init__.py]: Tests added :)

**Comment by bennuttall** [comment]
I've uncommented it to see what happens (tests will automatically run on travis). If they're failing on there we can look and see why. If they're passing, but not on your machine, let's look into that.

**Comment by bennuttall** [comment]
Ok got it. Pins 4 and 5 have been set up elsewhere with different functions and I guess the pin objects not closed / reset. If you can figure out how to do that, go for it. Otherwise @waveform80 or I will look when we get chance.

> `ValueError: pin 5 is already in use as a MockTriggerPin`

> `ValueError: pin 4 is already in use as a MockChargingPin`

Thank for the effort :+1:

File: gpiozero/boards.py [hunk 3]
```diff
@@ -1684,3 +1684,63 @@ def off(self):
         Turns the socket off.
         """
         self.value = False
+        
+class PumpkinPi(LEDBoard):
+    """
+    Extends :class: `LEDBoard` for the `ModMyPi Pumpkin Pi` board.
+
+    There are tweleve LEDs connected up to individual pins, so for
```
Comment (lurch) [inline gpiozero/boards.py]: tweleve -> twelve

File: gpiozero/boards.py [hunk 4]
```diff
@@ -1684,3 +1684,63 @@ def off(self):
         Turns the socket off.
         """
         self.value = False
+        
+class PumpkinPi(LEDBoard):
```
Comment (lurch) [inline gpiozero/boards.py]: Should be 2 blank lines before a new class

**Comment by tuftii** [comment]
Thanks @lurch for the fix! :+1:

**Comment by lurch** [comment]
@bennuttall / @waveform80 Did my 'fix' above to the MockFactory `pin_class` stuff get picked up? Or an equivalent fix committed?

**Pull request merged**